auth/services/oauth/oauth.go

```diff
@@ -155,6 +155,29 @@ func (s *service) validateActor(context *validationContext) error {
 	return nil
 }
 
+// check if the aud service identifier matches the oauth endpoint of the requested scope
+func (s *service) validateAud(context *validationContext) error {
+	if len(context.jwtBearerToken.Audience()) != 1 {
+		return errors.New("aud does not contain a single URI")
+	}
+	audience := context.jwtBearerToken.Audience()[0]
+	scope := context.jwtBearerTokenClaims.Scope
+	// parsing is already done in a previous check
+	subject, _ := did.ParseDID(context.jwtBearerToken.Subject())
+	iat := context.jwtBearerToken.IssuedAt()
+
+	uri, _, err := services.ResolveCompoundServiceURL(s.didResolver, *subject, scope, services.OAuthEndpointType, &iat)
```

Comment:
when we start supporting multiple current version, should this fail with an error? If so, all is OK, if not we might want to add a TODO here

Reply:
`ResolveCompoundServiceURL` already returns an error if `len(services) != 1`